### PR TITLE
Scaffold upgrade series logic 

### PR DESCRIPTION
**Pull request by SimonRichardson**
This is the first step at extracting the logic from state into the API
command pattern. It attempts to push down the business logic below the
facade level and push up the logic from state. That way it can be better
tested.

With this change we can start to implement (in following PRs) the
ability to query the charmhub API when validating the applications for
the correct series.

## QA steps

Tests should pass and nothing should have changed. Mostly logic
moving.

### CODE REVIEW

File: apiserver/facades/client/machinemanager/upgrade_series.go [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+		}
+		results[i] = app
+	}
+	return results, nil
+}
+
+type upgradeSeriesValidator struct{}
+
+func (s upgradeSeriesValidator) ValidateSeries(requested, machine, tag string) error {
+	return validateSeries(requested, machine, tag)
```
Comment (manadart) [inline apiserver/facades/client/machinemanager/upgrade_series.go:223]: `validateSeries` is only called by this method and is in _machinemanager.go_.
I suppose you'll reorganise this when you change it to call out to CharmHub...

**Review by manadart — APPROVED** [review]
Looks good.

File: apiserver/facades/client/machinemanager/upgrade_series.go (hunk 1 above)
Comment (SimonRichardson) [reply apiserver/facades/client/machinemanager/upgrade_series.go:223]: Indeed I have.

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**